sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -1479,3 +1479,20 @@ def test_issue_8653():
 def test_issue_9157():
     n = Symbol('n', integer=True, positive=True)
     atan(n - 1).is_nonnegative is True
+
+
+def test_trig_period():
+    x, y = symbols('x, y')
+
+    assert sin(x).period() == 2*pi
+    assert cos(x).period() == 2*pi
+    assert tan(x).period() == pi
+    assert cot(x).period() == pi
+    assert sec(x).period() == 2*pi
+    assert csc(x).period() == 2*pi
+    assert sin(2*x).period() == pi
+    assert cot(4*x - 6).period() == pi/4
+    assert cos((-3)*x).period() == 2*pi/3
+    assert cos(x*y).period(x) == 2*pi/abs(y)
```

Comment:
The reason this doesn't work with `periodicity` function is due to the check [here](https://github.com/sympy/sympy/pull/11277/files/c72047565116282ed35a42916559bb6d646643fc#diff-a5d2575656fef33ceed5fc4cd005e3c0R303)

```python
In []: f = cos(x*y)
In []: f.subs(x, x + 2*pi/Abs(y))
Out[]: cos(y*(x + 2*pi/Abs(y)))
```
As this cannot be simplified to `cos(x*y)`, I have raised a `NotImplementedError`.